universal-application-tool-0.0.1/app/views/applicant/ProgramIndexView.java

```diff
@@ -0,0 +1,54 @@
+package views.applicant;
+
+import static j2html.TagCreator.body;
+import static j2html.TagCreator.div;
+import static j2html.TagCreator.each;
+import static j2html.TagCreator.h1;
+import static j2html.TagCreator.h2;
+import static j2html.TagCreator.p;
+
+import com.google.common.collect.ImmutableList;
+import j2html.tags.Tag;
+import javax.inject.Inject;
+import play.twirl.api.Content;
+import services.program.ProgramDefinition;
+import views.BaseHtmlView;
+
+/** Returns a list of programs that an applicant can browse, with buttons for applying. */
+public class ProgramIndexView extends BaseHtmlView {
+
+  private final ApplicantLayout layout;
+
+  @Inject
+  public ProgramIndexView(ApplicantLayout layout) {
+    this.layout = layout;
+  }
+
+  /**
+   * For each program in the list, render the program information along with an "Apply" button that
+   * redirects the user to that program's application.
+   *
+   * @param programs an {@link ImmutableList} of {@link ProgramDefinition}s with the most recent
+   *     published versions
+   * @return HTML content for rendering the list of available programs
+   */
+  public Content render(long applicantId, ImmutableList<ProgramDefinition> programs) {
+    return layout.render(
+        body()
+            .with(h1("Programs"))
+            .with(each(programs, program -> shortProgram(applicantId, program))));
+  }
+
+  private Tag shortProgram(long applicantId, ProgramDefinition program) {
+    return div()
+        .with(h2(program.name()))
+        .with(p(program.description()))
+        .with(
+            redirectButton(
```

Comment:
Suggestion (blocking): use an `a` tag instead of this method. If we need it to look like a button we can style it as such later. `redirectButton` button isn't accessible so I'd like to remove it eventually.

Reply:
Thanks! Learned something new :) This is now an a tag